File: solutions_5688567749672960_0/C++/mikrosp1/A-small-0.cpp
#include<fstream>
#include<iostream>
#include<vector>

using namespace std;
int REVERSE(int n){
    int rev=0;
       while (n != 0)
   {
      rev = rev * 10;
      rev = rev + n%10;
      n       = n/10;
   }
   return rev;
}
int main(){
    int t,n;
    ifstream fin("A-small-0.in");
    ofstream fout("A-small-0.out");
    int dp[1000002];
    dp[0]=0;
    for(int i=1;i<=1000001;i++){
        int rev=0;
        n=i;
          while (n != 0)
   {
      rev = rev * 10;
      rev = rev + n%10;
      n       = n/10;
   }
   //cout<<i<<" "<<rev<<endl;
   if(rev>=i)dp[rev]=1000001;
   if(REVERSE(rev)!=i)rev=i-1;
        dp[i]=min(dp[i-1],dp[rev])+1;
   //       cout<<dp[i]<<" "<<rev<<" "<<i<<" "<<dp[i-1]<<endl;
    }



    fin>>t;
    for(int e=0;e<t;e++){
        fin>>n;
        fout<<"Case #"<<e+1<<": "<<dp[n]<<'\n';
    }
    fin.close();
    fout.close();
    return 0;
}

File: solutions_5688567749672960_0/Python/bbhands/A-small-0.py
#!/usr/bin/python
# -*- coding: utf-8 -*-

def rev(n):
    return int("".join(list(str(n))[::-1]))

if __name__ == "__main__":

    fin = open("A-small-0.in", "r")
    fout = open("A-small-0.out", "w")
    T = int(fin.readline())

    f = {}
    f[1] = 1
    for i in xrange(2, 10**6+1):
        r_i = rev(i)
        if r_i < i and len(str(i)) == len(str(r_i)):
            f[i] = min(f[i-1], f[r_i]) + 1
        else:
            f[i] = f[i-1] + 1

    #print f

    for t in xrange(0, T):
        N = int(fin.readline().strip())

        answer = f[N]

        fout.write("Case #%i: %s\n" % (t+1, answer))

    fin.close()
    fout.close()

